core/root/cluster.go

```diff
@@ -471,6 +471,29 @@ func (c clusterImpl) tags() map[string]string {
 func (c clusterImpl) Update(targets OperationTargets) (string, error) {
 	cfSvc := cloudformation.New(c.session)
 
+	exists, err := cfnstack.StackExists(cfSvc, c.controlPlane.ClusterName)
+	if err != nil {
+		logger.Errorf("please check your AWS Credentials/Permissions")
+		return "", fmt.Errorf("can't lookup AWS CloudFormation stacks")
+	}
+	if !exists {
+		logger.Errorf("you can only 'update' clusters with a matching cloudformation stack")
+		return "", fmt.Errorf("missing cluster root stack %s", c.controlPlane.ClusterName)
+	}
+
+	exists, err = cfnstack.NestedStackExists(cfSvc, c.controlPlane.ClusterName, naming.FromStackToCfnResource(c.etcd.Etcd.LogicalName()))
+	if err != nil {
+		logger.Errorf("please check your AWS Credentials/Permissions")
+		return "", fmt.Errorf("can't lookup AWS CloudFormation stacks")
+	}
+	// fail fast if it looks like we are trying to update a legacy cluster.
+	if !exists {
+		logger.Errorf("the %s stack must exist in order to be able to update your cluster.", naming.FromStackToCfnResource(c.etcd.Etcd.LogicalName()))
+		logger.Error("we're sorry, but kube-aws can not presently upgrade your cluster to the new release with a separate ETCD Cloudformation stack.")
+		logger.Error("please update using the previous 0.9.x branch or consider backing up, destroying and recreating to upgrade.")
```

Comment:
`0.10.x` instead of `0.9.x`? But anyway, I'd suggest to say this without mentioning a specific kube-aws ver. like `use a previous version of kube-aws in order to update, or consider creating an another cluster for migration.`